### ISSUE TITLE
authbuf did not hash correctly -- ensure key files are synced

### ISSUE THREAD

**Issue by slovichon** [pscedu/slash2#395]
On 2014-07-16 13:17:24 -0400, Zhihui Zhang wrote:

> ```
> On an archive file system, I was doing selfbuild + tar over ssh in parallel.
> 
> Thing goes pretty well for a long time until I hit the following:
> 
> (gdb) down
> # 5  0x000000000059c486 in slrpc_bulk_check (rq=0x8d2280, hbuf=0x7fff10001200, iov=0x7fff10001140, n=1) at /home/zhihui/projects-lime/slash_nara/share/rpc_common.c:1097
> 1097                    DEBUG_REQ(PLL_FATAL, rq, "authbuf did not hash "
> (gdb) l
> 1092            char tbuf[AUTHBUF_ALGLEN];
> 1093            int rc = 0;
> 1094
> 1095            slrpc_bulk_sign(rq, tbuf, iov, n);
> 1096            if (memcmp(tbuf, hbuf, AUTHBUF_ALGLEN)) {
> 1097                    DEBUG_REQ(PLL_FATAL, rq, "authbuf did not hash "
> 1098                        "correctly -- ensure key files are synced");
> 1099                    rc = SLERR_AUTHBUF_BADHASH;
> 1100            }
> 1101            return (rc);
> (gdb) p *rq
> $2 = {rq_type = 4711, rq_status = 0, rq_retry = 0, rq_timeout = 60, rq_request_portal = 30, rq_reply_portal = 31, rq_nob_received = 192, rq_reqlen = 264, rq_replen = 192, rq_import_generation = 0,
>   rq_sent = 1405530250, rq_lock = {psl_value = {value32 = 2}, psl_flags = 0, psl_owner_lineno = 0, psl_owner_file = 0x0, {<No data fields>}, psl_owner = 0}, rq_transno = 0, rq_xid = 6442597,
>   rq_history_seq = 0, rq_intr = 0, rq_replied = 1, rq_err = 0, rq_timedout = 0, rq_resend = 0, rq_restart = 0, rq_replay = 0, rq_no_resend = 0, rq_waiting = 0, rq_receiving_reply = 0, rq_no_delay = 0,
>   rq_net_err = 0, rq_abort_reply = 0, rq_timeoutable = 0, rq_bulk_abortable = 1, rq_refcount = {value32 = 1}, rq_retries = {value32 = 0}, rq_peer = {nid = 562995062530842, pid = 54321}, rq_self = 0,
>   rq_phase = PSCRPC_RQ_PHASE_COMPLETE, rq_send_state = PSCRPC_IMP_NOOP, rq_lentry = {plh_next = 0x0, plh_prev = 0x0, plh_magic = 1311693406324658740, plh_owner = 0x0}, rq_set_chain_lentry = {
>     plh_next = 0x8c6170, plh_prev = 0x8c6170, plh_magic = 1311693406324658740, plh_owner = 0x8c6170}, rq_history_lentry = {plh_next = 0x0, plh_prev = 0x0, plh_magic = 1311693406324658740,
>     plh_owner = 0x0}, rq_import = 0x8c77b0, rq_pool = 0x0, rq_conn = 0x0, rq_export = 0x0, rq_svc_thread = 0x0, rq_set = 0x8c6170, rq_reqmsg = 0x7fff10000f40, rq_repmsg = 0x7fff10001160, rq_req_cbid = {
>     cbid_fn = 0x48d0a2 <pscrpc_request_out_callback>, cbid_arg = 0x8d2280}, rq_reply_cbid = {cbid_fn = 0x48fec7 <pscrpc_reply_in_callback>, cbid_arg = 0x8d2280}, rq_bulk = 0x7fff10001050,
>   rq_interpret_reply = 0x528874 <msl_read_cb0>, rq_async_args = {space = {0, 0, 0, 0}, pointer_arg = {0x7fff10000e70, 0xeb24f0, 0x0, 0xe46b50, 0x0, 0x0, 0x0, 0x0, 0x0}}, rq_req_md_h = {
>     cookie = 105560241}, rq_compl = 0x0, rq_waitq = 0x0, rq_reply_md_h = {cookie = 105560237}, rq_reply_waitq = {wq_mut = {pm_mutex = {__data = {__lock = 0, __count = 0, __owner = 0, __nusers = 0,
>           __kind = 2, __spins = 0, __list = {__prev = 0x0, __next = 0x0}}, __size = '\000' <repeats 16 times>, "\002", '\000' <repeats 22 times>, __align = 0}, pm_owner = 0, pm_debug = 0}, wq_cond = {
>       __data = {__lock = 0, __futex = 0, __total_seq = 0, __wakeup_seq = 0, __woken_seq = 0, __mutex = 0x0, __nwaiters = 0, __broadcast_seq = 0}, __size = '\000' <repeats 47 times>, __align = 0},
>     wq_nwaiters = {value32 = 0}}, rq_arrival_time = {tv_sec = 0, tv_usec = 0}, rq_reply_state = 0x0, rq_rqbd = 0x0, rq_peer_qlen = 0x0}
> (gdb) p rc
> $3 = 0
> (gdb) bt
> # 0  0x00007ffff66d38e5 in raise () from /lib64/libc.so.6
> # 1  0x00007ffff66d50c5 in abort () from /lib64/libc.so.6
> # 2  0x00000000005f5e0a in pfl_abort () at /home/zhihui/projects-lime/pfl/init.c:121
> # 3  0x000000000060d5fc in _psclogv (pci=0x7fff10000950, level=0, options=0, 
>     fmt=0x656f08 "req@%p x%ld/t%ld cb=%p c%lx o%d->@%s:%d lens %d/%d ref %d res %d ret %d fl %s:%s%s%s%s%s%s%s%s%s%s%s/%x/%x replyc %lx rc %d/%d to=%d sent=%ld :: authbuf did not hash correctly -- ensure key files are "..., ap=0x7fff1f5f9c60) at /home/zhihui/projects-lime/pfl/log.c:430
> # 4  0x000000000060d7f3 in _psc_fatal (pci=0x7fff10000950, level=0, options=0, 
>     fmt=0x656f08 "req@%p x%ld/t%ld cb=%p c%lx o%d->@%s:%d lens %d/%d ref %d res %d ret %d fl %s:%s%s%s%s%s%s%s%s%s%s%s/%x/%x replyc %lx rc %d/%d to=%d sent=%ld :: authbuf did not hash correctly -- ensure key files are "...) at /home/zhihui/projects-lime/pfl/log.c:463
> # 5  0x000000000059c486 in slrpc_bulk_check (rq=0x8d2280, hbuf=0x7fff10001200, iov=0x7fff10001140, n=1) at /home/zhihui/projects-lime/slash_nara/share/rpc_common.c:1097
> # 6  0x000000000040fcbe in authbuf_check (rq=0x8d2280, msgtype=4713) at /home/zhihui/projects-lime/slash_nara/share/authbuf_sign.c:181
> # 7  0x0000000000528a18 in msl_read_cb0 (rq=0x8d2280, args=0x8d23e0) at /home/zhihui/projects-lime/slash_nara/mount_slash/mount_slash_int.c:809
> # 8  0x000000000059ff4f in pscrpc_interpret (rq=0x8d2280) at /home/zhihui/projects-lime/pfl/rpcclient.c:134
> # 9  0x00000000005bbe41 in pscrpc_check_set (set=0x8c6170, check_allsent=1) at /home/zhihui/projects-lime/pfl/rpcclient.c:1004
> # 10 0x00000000005bf6c2 in pscrpc_set_wait (set=0x8c6170) at /home/zhihui/projects-lime/pfl/rpcclient.c:1256
> # 11 0x0000000000537412 in msl_pages_prefetch (r=0xe46b50) at /home/zhihui/projects-lime/slash_nara/mount_slash/mount_slash_int.c:1307
> # 12 0x000000000054491a in msl_io (pfr=0x150fe80, mfh=0xe73220, buf=0x7fff10003d90 "", size=1605, off=0, rw=SL_READ) at /home/zhihui/projects-lime/slash_nara/mount_slash/mount_slash_int.c:1812
> # 13 0x00000000005067ae in mslfsop_read (pfr=0x150fe80, size=4096, off=0, data=0xe73220) at /home/zhihui/projects-lime/slash_nara/mount_slash/main.c:2726
> # 14 0x00000000004a8e98 in pscfs_fuse_handle_read (req=0x7fff10000de0, inum=327636872897997014, size=4096, off=0, fi=0x7fff1f5fdac0) at /home/zhihui/projects-lime/pfl/fuse.c:933
> # 15 0x00007ffff79338e0 in do_read (req=<value optimized out>, nodeid=<value optimized out>, inarg=<value optimized out>) at fuse_lowlevel.c:1249
> # 16 0x00007ffff793484d in fuse_ll_process_buf (data=0x8a10a0, buf=0x7fff1f5fdc60, ch=0x8a13b8) at fuse_lowlevel.c:2458
> # 17 0x00007ffff7934c23 in fuse_ll_process (data=<value optimized out>, buf=<value optimized out>, len=<value optimized out>, ch=<value optimized out>) at fuse_lowlevel.c:2480
> # 18 0x00000000004a28d4 in pscfs_fuse_listener_loop (arg=0x0) at /home/zhihui/projects-lime/pfl/fuse.c:364
> # 19 0x00007ffff6a3b851 in start_thread () from /lib64/libpthread.so.0
> # 20 0x00007ffff678994d in clone () from /lib64/libc.so.6
> (gdb)
> ```
> 
> On 2014-07-16 13:19:07 -0400, Zhihui Zhang wrote:
> 
> ```
> 
> zhihui@lime: ~/projects-lime/slash_nara/mount_slash$ svn info
> Path: .
> URL: svn+ssh://frodo/cluster/svn/projects/slash_nara/mount_slash
> Repository Root: svn+ssh://frodo/cluster/svn
> Repository UUID: 3eda493b-6a19-0410-b2e0-ec8ea4dd8fda
> Revision: 24098
> Node Kind: directory
> Schedule: normal
> Last Changed Author: zhihui
> Last Changed Rev: 24098
> Last Changed Date: 2014-07-16 10:56:08 -0400 (Wed, 16 Jul 2014)
> ```
> 
> On 2014-08-09 02:58:41 -0400, Jared Yanovich wrote:
> 
> ```
> 
> 
> *** This bug has been marked as a duplicate of bug 372 ***
> ```
